setup.py

```diff
@@ -15,6 +15,7 @@
 from setuptools import setup, find_packages
 import pathlib
 from itertools import chain
+from distutils.util import strtobool
```

Comment:
I would like to avoid `distutils` since the package is deprecated in python 3.10 and will be removed completely in 3.12. 

I'm open to using any of our existing dependencies or writing a quick utility function to mimic this behavior.
relevant links:
https://peps.python.org/pep-0632/#migration-advice
https://stackoverflow.com/questions/77941994/what-is-setuptools-alternative-to-the-deprecated-distutils-strtobool
https://danielms.site/zet/2023/pythons-distutil-strtobool-replacement/